File: Python/unit_root_sim.py
# -*- coding: utf-8 -*-
"""
Created on Thu Nov 29 22:40:02 2018

@author: Zhentao
"""

# unit root


#%%
import numpy as np


def AR(a,T):
    y = np.zeros(T)
    for t in range(1,T):
        y[t] = a * y[t-1] + np.random.randn(1)
    return y


#%%
T = 100
a = 1.0

Rep = 1000
Bhat = np.zeros(Rep)

for r in range(Rep):
    Y = AR(a,T)
    y = Y[1:(T+1)]; y = np.reshape(y,(-1,1))
    x = Y[range(T-1)]; x = np.reshape(x,(-1,1))
    bhat = np.linalg.lstsq(x , y, rcond=None)[0]
    Bhat[r] = bhat

#%%

# using PyPlot

import matplotlib.pyplot as plt
import seaborn as sns

sns.distplot(Bhat)

File: Python/unit_root_sim.jl
# unit root


using Distributions

function AR(a::Float64,T::Int)
    y = zeros(T,1)
    for t in 2:T
        y[t] = a * y[t-1] + randn(1)[1]
    end
    return y
end


T = 100
a = 1.0


Rep = 1000
Bhat = zeros(Float64,Rep)

for r in 1:Rep
    Y = AR(a,T)
    y = Y[2:T]
    x = Y[1:(T-1)]
    bhat = x \ y
    Bhat[r] = bhat
end


# using PyPlot

using PyPlot
using Gadfly
using KernelDensity

Gadfly.plot(Bhat, x = 0.0:0.01:1.5, Geom.density)
